crates/oxc_linter/src/rules/import/no_dynamic_require.rs

```diff
@@ -0,0 +1,127 @@
+use oxc_ast::{ast::Expression, AstKind};
+use oxc_diagnostics::OxcDiagnostic;
+use oxc_macros::declare_oxc_lint;
+use oxc_span::{GetSpan, Span};
+
+use crate::{context::LintContext, rule::Rule, AstNode};
+
+fn no_dnyamic_require_diagnostic(span: Span) -> OxcDiagnostic {
+    OxcDiagnostic::warn("Expected a literal string or immutable template literal")
+        .with_help("Replace the argument with a literal string or immutable template literal")
+        .with_label(span)
+}
+
+#[derive(Debug, Default, Clone)]
+pub struct NoDynamicRequire {
+    esmodule: bool,
+}
+
+declare_oxc_lint!(
+    /// ### What it does
+    ///
+    /// The require method from CommonJS is used to import modules from different files. Unlike the
+    /// ES6 import syntax, it can be given expressions that will be resolved at runtime. While this
+    /// is sometimes necessary and useful, in most cases it isn't. Using expressions (for instance,
+    /// concatenating a path and variable) as the argument makes it harder for tools to do static
```

Comment:
Please follow the standard docs template.

Reply:
This is not resolved. Your example sections do not match the template. Please try again.

Reply:
Reminder that we use these comments to generate documentation pages on our website. This is directly user facing and should be high quality.

Reply:
Do you mean the missing space between the Bad and Good example?

Reply:
https://github.com/oxc-project/oxc/blob/1d3e973f101fc2d03f705f75c7de9e9cb04ae50a/tasks/rulegen/template.txt#L24

Reply:
Alright sorry that I missed that, the import rules have no rule generator so I copied one with the old template. Seems there are quite some rules which should be converted.